File: config/src/artifact/vorpal_job.rs
use crate::artifact::{vorpal::Vorpal, SYSTEMS};
use anyhow::Result;
use vorpal_sdk::{
    artifact::{get_env_key, Job},
    context::ConfigContext,
};

#[derive(Default)]
pub struct VorpalJob {}

impl VorpalJob {
    pub fn new() -> Self {
        Self::default()
    }

    pub async fn build(self, context: &mut ConfigContext) -> Result<String> {
        let vorpal = Vorpal::new().build(context).await?;

        let script = format!("{}/bin/vorpal --version", get_env_key(&vorpal));

        Job::new("vorpal-job", script, SYSTEMS.to_vec())
            .with_artifacts(vec![vorpal.clone()])
            .build(context)
            .await
    }
}

File: sdk/go/cmd/vorpal/artifact/vorpal_job.go
package artifact

import (
	"fmt"

	"github.com/ALT-F4-LLC/vorpal/sdk/go/pkg/artifact"
	"github.com/ALT-F4-LLC/vorpal/sdk/go/pkg/config"
)

func BuildVorpalJob(context *config.ConfigContext) (*string, error) {
	vorpal, err := BuildVorpal(context)
	if err != nil {
		return nil, fmt.Errorf("failed to build vorpal: %w", err)
	}

	script := fmt.Sprintf("\n%s/bin/vorpal --version", artifact.GetEnvKey(vorpal))

	return artifact.NewTask("vorpal-job", script, SYSTEMS).
		WithArtifacts([]*string{vorpal}).
		Build(context)
}
